sdks/java/core/src/main/java/org/apache/beam/sdk/values/PCollectionView.java

```diff
@@ -50,12 +51,20 @@
 
   /**
    * For internal use only.
+   *
+   * @deprecated use getViewFn()
    */
+  @Deprecated
   public T fromIterableInternal(Iterable<WindowedValue<?>> contents);
 
   /**
    * For internal use only.
    */
+  public ViewFn<Iterable<WindowedValue<?>>, T> getViewFn();
```

Comment:
Should we even expose this?

We shouldn't add methods that pipeline authors wouldn't interact with. The pipeline construction code should note the type of view and store that information internally similar to how the DataflowRunner keeps track of pcollection views that require indexing.